.github/workflows/continuous-integration.yml

```diff
@@ -35,18 +35,20 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
       - name: Update NPM
+        if: matrix.node-version <= 14
```

Comment:
I believe the NPM update is necessary as failing tests were not causing a pipeline failure.

Reply:
The tests do not fail because of the NPM update. We should let it install different versions of NPM but always be above v7.